msal/token_cache.py

```diff
@@ -126,7 +126,13 @@ def _get(self, credential_type, key, default=None):  # O(1)
 
     @staticmethod
     def _is_matching(entry: dict, query: dict, target_set: set = None) -> bool:
-        return is_subdict_of(query or {}, entry) and (
+        query_with_lowercase_environment = {
+            # __add() canonicalized entry's environment value to lower case,
+            # so we do the same here.
+            k: v.lower() if k == "environment" else v
+            for k, v in query.items()
```

Comment:
would be good to have some unit tests to assert retrieval with mixed-case query succeeds

Reply:
Thanks for the reminding. A new test case has been added below.

Reply:
@gladjohn , can you review this again to see whether your comments were resolved?